master/internal/provisioner/provisioner_test.go

```diff
@@ -309,3 +320,48 @@ func TestProvisionerNotProvisionExtraInstances(t *testing.T) {
 	assert.NilError(t, mock.system.StopAndAwaitTermination())
 	assert.DeepEqual(t, mock.cluster.history[len(mock.cluster.history)-1], newMockFuncCall("list"))
 }
+
+func TestProvisionerTerminateUnconnectedInstances(t *testing.T) {
+	setup := &mockConfig{
+		initTime:               time.Now().Add(-time.Hour),
```

Comment:
Is this still useful?

Reply:
Oh yeah in latest version it's not needed. It was needed before when weren't shutting down agents the first 5 minutes after the master started up.